posthog/hogql/property.py

```diff
@@ -549,6 +551,48 @@ def property_to_expr(
             ),
             right=ast.Constant(value=cohort.pk),
         )
+    elif property.type == "event_metadata":
+        if scope != "event":
+            raise NotImplementedError(f"The '{property.type}' property filter does not work in '{scope}' scope")
+        operator = cast(Optional[PropertyOperator], property.operator) or PropertyOperator.EXACT
+        value = property.value
+
+        if isinstance(value, list):
+            if len(value) == 0:
+                return ast.Constant(value=1)
+            elif len(value) == 1:
+                value = value[0]
+            else:
+                # Using an AND here instead of `in()` or `notIn()`, due to Clickhouses poor handling of `null` values
+                exprs = [
+                    property_to_expr(
+                        Property(
+                            type=property.type,
+                            key=property.key,
+                            operator=property.operator,
+                            value=v,
+                        ),
+                        team,
+                        scope,
+                    )
+                    for v in value
+                ]
+                if (
+                    operator == PropertyOperator.NOT_ICONTAINS
+                    or operator == PropertyOperator.NOT_REGEX
+                    or operator == PropertyOperator.IS_NOT
+                ):
+                    return ast.And(exprs=exprs)
+                return ast.Or(exprs=exprs)
```

Comment:
This and/or looks like a duplicate of the above block with
```
        property.type == "event"
        or property.type == "feature"
        or property.type == "person"
        or property.type == "group"
        or property.type == "data_warehouse"
        or property.type == "data_warehouse_person_property"
        or property.type == "session"
        or property.type == "recording"
        or property.type == "log_entry"`
```

Could this logic just be merged there instead?

Reply:
> Could this logic just be merged there instead?

Yeah, good catch. Updated with b68ac92ea9